Core/Graphics/Source/GraphicsImpl.cpp

```diff
@@ -169,79 +168,59 @@ namespace Babylon
-        }
+        m_safeTimespanGuarantor.EndSafeTimespan();
 
-        if (error != nullptr)
-        {
-            std::rethrow_exception(error);
-        }
+        Frame();
 
-        if (workDone)
-        {
-            {
-                std::scoped_lock lock{m_state.Mutex};
-                if (m_state.Bgfx.Dirty)
-                {
-                    bgfx::discard();
-                }
-            }
-
-            bgfx::frame();
-        }
-
-        auto oldRenderTaskCompletionSource = m_afterRenderTaskCompletionSource;
-        m_afterRenderTaskCompletionSource = {};
-        oldRenderTaskCompletionSource.complete();
+        m_afterRenderScheduler.m_dispatcher.tick(*m_cancellationSource);
 
         m_rendering = false;
     }
 
-    arcana::task<void, std::exception_ptr> Graphics::Impl::RenderCurrentFrameAsync(bool& finished, bool& workDone, std::exception_ptr& error)
+    Graphics::Impl::UpdateToken Graphics::Impl::GetUpdateToken()
     {
-        bool anyTasks{};
-        arcana::task<void, std::exception_ptr> whenAllTask{};
+        return {*this};
+    }
+
+    FrameBuffer& Graphics::Impl::AddFrameBuffer(bgfx::FrameBufferHandle handle, uint16_t width, uint16_t height, bool backBuffer)
+    {
+        if (!m_frameBufferManager)
         {
-            std::scoped_lock RenderWorkTasksLock{m_renderWorkTasksMutex};
-            anyTasks = !m_renderWorkTasks.empty();
-            if (anyTasks)
-            {
-                whenAllTask = arcana::when_all<std::exception_ptr>(m_renderWorkTasks);
-                m_renderWorkTasks.clear();
-            }
+            throw std::runtime_error{"FrameBufferManager has not been initialized!"};
         }
 
-        workDone = workDone || anyTasks;
+        return m_frameBufferManager->AddFrameBuffer(handle, width, height, backBuffer);
+    }
 
-        if (!anyTasks)
+    void Graphics::Impl::RemoveFrameBuffer(const FrameBuffer& frameBuffer)
+    {
+        if (!m_frameBufferManager)
         {
-            finished = true;
-            return arcana::task_from_result<std::exception_ptr>();
+            throw std::runtime_error{"FrameBufferManager has not been initialized!"};
         }
-        else
+
+        m_frameBufferManager->RemoveFrameBuffer(frameBuffer);
+    }
+
+    FrameBuffer& Graphics::Impl::DefaultFrameBuffer()
+    {
+        if (!m_frameBufferManager)
         {
-            return whenAllTask.then(m_renderWorkDispatcher, arcana::cancellation::none(), [this, &finished, &workDone, &error](const arcana::expected<void, std::exception_ptr>& result) mutable {
-                if (result.has_error())
-                {
-                    finished = true;
-                    error = result.error();
-                    return arcana::task_from_result<std::exception_ptr>();
-                }
-
-                return RenderCurrentFrameAsync(finished, workDone, error);
-            });
+            throw std::runtime_error{"FrameBufferManager has not been initialized!"};
```

Comment:
FrameBufferManager is created and destroyed on EnableRendering and DisableRendering. Even if we revived the `whenGraphicsReady` code, it can still happen.